src/LanguageServer/Protocol/Workspaces/LspMiscellaneousFilesWorkspaceProvider.cs

```diff
@@ -3,27 +3,130 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Features.Workspaces;
 using Microsoft.CodeAnalysis.Host;
-using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.LanguageServer.Handler;
 using Microsoft.CodeAnalysis.MetadataAsSource;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.CommonLanguageServerProtocol.Framework;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer;
 
 /// <summary>
-/// Service to create <see cref="LspWorkspaceManager"/> instances.
-/// This is not exported as a <see cref="ILspServiceFactory"/> as it requires
-/// special base language server dependencies such as the <see cref="HostServices"/>
+/// Defines a default workspace for opened LSP files that are not found in any
+/// workspace registered by the <see cref="LspWorkspaceRegistrationService"/>.
+/// If a document added here is subsequently found in a registered workspace, 
+/// the document is removed from this workspace.
+/// 
+/// Future work for this workspace includes supporting basic metadata references (mscorlib, System dlls, etc),
+/// but that is dependent on having a x-plat mechanism for retrieving those references from the framework / sdk.
 /// </summary>
-[ExportCSharpVisualBasicStatelessLspService(typeof(LspMiscellaneousFilesWorkspaceProvider)), Shared]
-[method: ImportingConstructor]
-[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class LspMiscellaneousFilesWorkspaceProvider(IMetadataAsSourceFileService metadataAsSourceFileService) : ILspService
+internal sealed class LspMiscellaneousFilesWorkspaceProvider(ILspServices lspServices, IMetadataAsSourceFileService metadataAsSourceFileService, HostServices hostServices)
+    : Workspace(hostServices, WorkspaceKind.MiscellaneousFiles), ILspMiscellaneousFilesWorkspaceProvider, ILspWorkspace
```

Comment:
being a misc workspace provider *and* a workspace at the same time is breaking my brain.